File: storage/db-core/src/main/java/com/coffee/gu/store/SalesHourEntity.java
package com.coffee.gu.store;

import com.coffee.gu.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

import java.time.DayOfWeek;
import java.time.LocalTime;

@Table(name = "sales_hour")
@Entity
public class SalesHourEntity extends BaseEntity {
    private Long storeId;
    @Column(name = "`day`")
    private DayOfWeek day;
    private LocalTime open;
    private LocalTime close;

    public SalesHourEntity() {}

    public SalesHourEntity(Long storeId, DayOfWeek day, LocalTime open, LocalTime close) {
        this.storeId = storeId;
        this.day = day;
        this.open = open;
        this.close = close;
    }

    public SalesHour toModel() {
        return new SalesHour(
                this.storeId,
                this.day,
                this.open,
                this.close
        );
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/store/SalesHourEntity.kt
package com.coffee.gu.store

import com.coffee.gu.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.DayOfWeek
import java.time.LocalTime

@Table(name = "sales_hour")
@Entity
class SalesHourEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val storeId: Long,
    @Column(name = "`day`")
    val day: DayOfWeek,
    val open: LocalTime,
    val close: LocalTime,
) : BaseEntity() {

    fun toModel(): SalesHour {
        return SalesHour(
            storeId = storeId,
            day = day,
            open = open,
            close = close,
        )
    }
}
